src/legacy/core_plugins/kbn_vislib_vis_types/public/editors/metrics_axes_options.tsx

```diff
@@ -0,0 +1,316 @@
+      };
+      setValue('seriesParams', series);
+    },
+    [stateParams.seriesParams, setValue]
+  );
+
+  const updateAxisTitle = () => {
+    let lastMatchingType = lastMatchingSeriesAggType;
+    let lastMatchingFeild = lastMatchingSeriesAggField;
+    const lastLabels = lastCustomLabels;
+
+    stateParams.valueAxes.forEach((axis, axisNumber) => {
+      let newCustomLabel = '';
+      const isFirst = axisNumber === 0;
+      const matchingSeries: AggConfig[] = [];
+
+      stateParams.seriesParams.forEach((series, seriesIndex) => {
+        if ((isFirst && !series.valueAxis) || series.valueAxis === axis.id) {
+          let seriesNumber = 0;
+          aggs.forEach((agg: AggConfig) => {
+            if (agg.schema.name === 'metric') {
+              if (seriesNumber === seriesIndex) {
+                matchingSeries.push(agg);
+              }
+              seriesNumber++;
+            }
+          });
+        }
+      });
+
+      if (matchingSeries.length === 1) {
+        newCustomLabel = matchingSeries[0].makeLabel();
+      }
+
+      const matchingSeriesAggType = get(matchingSeries, '[0]type.name', '');
+      const matchingSeriesAggField = get(matchingSeries, '[0]params.field.name', '');
+
+      if (lastCustomLabels[axis.id] !== newCustomLabel && newCustomLabel !== '') {
+        const aggTypeIsChanged = lastMatchingSeriesAggType !== matchingSeriesAggType;
+        const aggFieldIsChanged = lastMatchingSeriesAggField !== matchingSeriesAggField;
+        const aggIsChanged = aggTypeIsChanged || aggFieldIsChanged;
+        const lastCustomLabelMatchesAxisTitle = lastCustomLabels[axis.id] === axis.title.text;
+        const isFirstRender = Object.keys(lastCustomLabels).length === 0;
+
+        if (
+          !isFirstRender &&
+          (aggIsChanged || axis.title.text === '' || lastCustomLabelMatchesAxisTitle)
+        ) {
+          // Override axis title with new custom label
+          setValueAxisByIndex(axisNumber, 'title', { ...axis, text: newCustomLabel });
+        }
+        lastLabels[axis.id] = newCustomLabel;
+      }
+      lastMatchingType = matchingSeriesAggType;
+      lastMatchingFeild = matchingSeriesAggField;
+    });
+
+    setLastCustomLabels(lastLabels);
+    setLastMatchingSeriesAggType(lastMatchingType);
+    setLastMatchingSeriesAggField(lastMatchingFeild);
+  };
+
+  const getUpdatedAxisName = useCallback(
+    (axisPosition: ValueAxis['position']) => {
+      const axisName = capitalize(axisPosition) + AXIS_PREFIX;
+      let lastAxisNameNumber = axesNumbers[axisPosition];
+
+      if (!lastAxisNameNumber) {
+        lastAxisNameNumber = stateParams.valueAxes.reduce(countNextAxisNumber(axisName, 'name'), 0);
+      }
+      const nextAxisNameNumber = lastAxisNameNumber + 1;
+      setAxesNumbers({ ...axesNumbers, [axisPosition]: nextAxisNameNumber });
+
+      return `${axisName}${nextAxisNameNumber}`;
+    },
+    [stateParams.valueAxes, axesNumbers, setAxesNumbers, countNextAxisNumber]
```

Comment:
```suggestion
    [stateParams.valueAxes, axesNumbers]
```
There is no necessity to include static functions in this array